components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListener.java

```diff
@@ -100,6 +102,14 @@ public boolean doPreUpdateApplication(ServiceProvider serviceProvider, String te
         handleOAuthAppAssociationRemoval(serviceProvider);
         storeSaaSPropertyValue(serviceProvider);
         removeClientSecret(serviceProvider);
+
+        ServiceProvider existingSP = OAuth2ServiceComponentHolder.getApplicationMgtService()
+                .getServiceProvider(serviceProvider.getApplicationID());
+        String newClaimConfigString = new Gson().toJson(serviceProvider.getClaimConfig());
```

Comment:
If the order get changed, string comparison will be failed. Therefore, compare two objects.